src/useAsyncFn.ts

```diff
@@ -18,17 +18,23 @@ export type AsyncState<T> =
       value: T;
     };
 
-const useAsyncFn = <T>(fn: (...args: any[]) => Promise<T>, deps: DependencyList = []): [AsyncState<T>, () => void] => {
-  const [state, set] = useState<AsyncState<T>>({
-    loading: false,
-  });
+export default function useAsyncFn<
+  Result = any,
+  Args extends any[] = any[],
+  Fn extends Function = (...args: Args | []) => Promise<Result>
+>(
+  fn: Fn,
+  deps: DependencyList = [],
+  initialState: AsyncState<Result> = { loading: false }
+): [AsyncState<Result>, (...args: Args | []) => Promise<Result>] {
+  const [state, set] = useState<AsyncState<Result>>(initialState);
 
   const mounted = useRefMounted();
 
-  const callback = useCallback((...args) => {
+  const callback = useCallback((...args: Args | []) => {
     set({ loading: true });
 
-    fn(...args).then(
+    return fn(...args).then(
```

Comment:
nice

What is you use case for returning value? Also I think, if you want to return it here, you also need to return it from the `.then(value => { return value; })`.

Reply:
mostly so we can await the callback

```ts
const [state, callback] = ...

// ...

//ideally,
const myReturnValue = await callback
// probably need to return value and return error in each of the promise callbacks though. hmm.

need more tests
```